sdk/azidentity/cache/cache.go

```diff
@@ -8,45 +8,99 @@
 package cache
 
 import (
+	"bytes"
+	"context"
 	"fmt"
 	"path/filepath"
+	"sync"
+	"time"
 
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
 	"github.com/Azure/azure-sdk-for-go/sdk/azidentity/internal"
 	extcache "github.com/AzureAD/microsoft-authentication-extensions-for-go/cache"
-	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/cache"
+	msal "github.com/AzureAD/microsoft-authentication-library-for-go/apps/cache"
 )
 
 const defaultName = "msal.cache"
 
-func init() {
-	internal.NewCache = func(o *internal.TokenCachePersistenceOptions, enableCAE bool) (cache.ExportReplace, error) {
-		if o == nil {
-			return nil, nil
+var (
+	// once ensures New tests the storage implementation only once
+	once = &sync.Once{}
+	// storageError is the error from the storage test
+	storageError error
```

Comment:
This effectively prevents retries if the first attempt fails. I suspect that this is ok as there shouldn't be any kind of transient error here (how would one even handle that?).

Reply:
That's my thinking as well. Some reasons for failure are:
1. can't write a file
2. can't unlock mac keychain
3. Linux kernel API unavailable/forbidden

So failures probably aren't transient. Even so, I guess a user could take some action outside the application that would make storage possible on retry, but that seems unlikely to actually happen and not worth the cost of `New` repeatedly testing the storage implementation in the success case (success is also probably not transient).